File: references/cellab_devkit-1.0 (1)/rules_java/border.java
/*

        The Vichniac voting rule with the seven
        extra bits used as memory

*/

class Border extends ruletable {

    void jcruleModes() {
        setPaletteRequest("mask1");
        setPatternRequest("square");
    }

    int jcrule(int oldstate) {
        int NineSum = nw + n + ne + e + se + s + sw + w + self,
            Cycle = (oldstate >> 7) & 1,
            NewCycle = Cycle ^ 1, NewSelf = 0;

        switch (Cycle) {
            case 0:
                NewSelf = (NineSum > 0) ? 1 : 0;
                break;

            case 1:
                NewSelf = (NineSum == 9) ? 0 : self;
                break;
        }

        return (NewCycle << 7) | NewSelf;
    }
}

public class border {
    public static void main(String args[]) {
        (new Border()).generateRuleFile("border");
    }
}

File: references/cellab_devkit-1.0 (1)/ruledefs/border.js
    /*

	This rule alternates between two cycles: In cycle 0,
	every cell touching a firing cell is turned on.  In
	cycle 1, any cell which is the center of a block of 9
	firing cells is turned off.  Bit #0 is the firing bit
	and bit #7 is the cycle bit.

    */

    rule.worldtype = 1;          // 2D torus world
    rule.patreq = "square";
    rule.palreq = "mask1";
    
    function border(oldstate,     nw, n  , ne,
                                  w, self, e,
                                  sw, s  , se) {
        var nineSum = nw + n + ne + w + self + e + sw + s + se, 
            cycle = (oldstate >> 7) & 1,
            newCycle = cycle ^ 1,
            newSelf = 0;

        switch (cycle) {
            case 0:
                newSelf = (nineSum > 0) ? 1 : 0;
                break;

            case 1:
                newSelf = (nineSum == 9) ? 0 : self;
                break;
        }

        return (newCycle << 7) | newSelf;
    }
